realm/realm-library/src/main/java/io/realm/MutableRealmInteger.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import io.realm.annotations.Beta;
+import io.realm.internal.ManagableObject;
+import io.realm.internal.Row;
+
+
+/**
+ * A MutableRealmInteger is a mutable, {@link java.lang.Long}-like numeric quantity.
+ * MutableRealmInteger implementations are not thread safe.  Like all Realm objects,
+ * managed RealmIntegers may not be moved across threads.  Unmanaged RealmObject
+ * may be moved across threads but require safe publication.
```

Comment:
Standard term.  "Java Concurrency in Practice", section 3.5.
tl;dr, just what it sounds like: you have to do some kind of synchronization to pass something to another thread.